### PR TITLE
Unit tests for WorkspaceEventLoop.

### PR DESCRIPTION
**Pull request by jparsai**
#### Description:
This PR will add more unit test cases for `backend/eventloop/workspace_event_loop.go`

##### JIRA Ticket:
https://issues.redhat.com/browse/GITOPSRVCE-113

### CODE REVIEW

File: backend/eventloop/workspace_event_loop_test.go [hunk 1]
```diff
@@ -69,15 +125,162 @@ var _ = Describe("Workspace Event Loop Test", func() {
...
+				Expect(result.Event).To(Equal(msg.Event))
+				Expect(result.MessageType).To(Equal(msg.MessageType))
+			}()
+
+			// Consider test case passed if application event loop is not created in 5 seconds.
+			Consistently(tAELF.numberOfEventLoopsCreated, "5s").Should(Equal(2))
+
+		})
+
+		It("Should unorphan previous GitOpsDeploymentSyncRun event if parent GitOpsDeployment event is passed and new application event loop should be created.", func() {
```
Comment (jparsai) [inline backend/eventloop/workspace_event_loop_test.go:228]: This test is failing.

File: backend/eventloop/workspace_event_loop_test.go [hunk 2]
```diff
@@ -69,15 +93,234 @@ var _ = Describe("Workspace Event Loop Test", func() {
...
+			Expect(deploymentModifiedMsg).NotTo(BeNil())
+			Expect(deploymentModifiedMsg.Event).To(Equal(msg.Event))
+			Expect(deploymentModifiedMsg.MessageType).To(Equal(msg.MessageType))
+
+			syncRunModifiedMsg := <-tAELF.outputChannel
 
+			// Make sure this is the gitopsdeploymentsyncrun event from above
+			Expect(syncRunModifiedMsg).NotTo(BeNil())
+			Expect(syncRunModifiedMsg.Event.Request.Name).To(Equal(msgTemp.Event.Request.Name))
+			Expect(deploymentModifiedMsg.MessageType).To(Equal(msgTemp.MessageType))
```
Comment (jgwest) [inline backend/eventloop/workspace_event_loop_test.go:314]: Guessing this line (314) is supposed to be `Expect(syncRunModifiedMsg)`? At the moment it's the same as a line above.

**Review by jgwest — CHANGES_REQUESTED** [review]

File: backend/eventloop/workspace_event_loop_test.go (hunk 2 above)
Comment (jparsai) [reply backend/eventloop/workspace_event_loop_test.go]: Yes, I missed to change it after copying from above :).

**Review by jgwest — APPROVED** [review]
Looks great, thanks @jparsai!

**Pull request merged**